File: Level-2/12.java
class Solution {
    
    public int check(int n) {
        for(int i = 2; i <= Math.sqrt(n); i++){
            if(n % i == 0 && n / i <= 10000000){
                return n / i;
            }
        }
        return 1;
    }
    
    public int[] solution(long begin, long end) {
        int b = (int)begin;
        int e = (int)end;
        
        int[] answer = new int[e - b + 1];
        
        for(int i = b; i <= e; i++){
            if(i == 1) answer[i - b] = 0;
            else answer[i - b] = check(i);
        }
        
        return answer;
    }
}

File: Level-2/12.js
const factorial = (num) => {
    let res = 1;
    for(let i = 2; i <= num; i++) {
        res *= i;
    }
    return res;
}

function solution(n, k) {
    const answer = [];
    const arr = Array.from(Array(n), (_, i) => i + 1);
    
    k--;
    while(arr.length) {
        if(k === 0) {
            answer.push(...arr);
            break;
        }
        
        const fact = factorial(arr.length - 1);
        const index = k / fact >> 0;
        k = k % fact;
        
        answer.push(arr[index]);
        arr.splice(index, 1);
    }
    
    return answer;
}
